ultraplot/axes/geo.py

```diff
@@ -825,35 +1122,109 @@ def _add_geoticks(self, x_or_y, itick, ticklen):
         if isinstance(gl, tuple):
             locator = gl[0] if x_or_y == "x" else gl[1]
             tick_positions = np.asarray(list(locator.keys()))
-            # Show the ticks but hide the labels
-            ax.set_ticks(tick_positions)
+            # Turn off the ticks otherwise they are double for
+            # basemap (different from cartopy)
             ax.set_major_formatter(mticker.NullFormatter())
 
+        else:
+            if x_or_y == "x":
+                lim = self._lonaxis.get_view_interval()
+                locator = gl.xlocator
+                tick_positions = self._lonaxis._get_ticklocs(locator)[1:-1]
+            else:
+                lim = self._lataxis.get_view_interval()
+                locator = gl.ylocator
+                tick_positions = self._lataxis._get_ticklocs(locator)[
+                    1:-1
+                ]  # cutoff edges
```

Comment:
Why are we truncating the edges by hand here? This is a hack to get a test to pass and not something that generalizes.

Reply:
Not sure will do some more digging as to why this is caused. I am setting the tick locs specifically, which has the range in the pic, but the first subplot (without the tics) clearly has a reduced range.

Reply:
please note that the reduced range is the target range that needs to be set so it is indeed a bug.

Reply:
Think I figured it out. The update extent method is called before the ticks are set. If I call it again if we add the ticks, it looks good for cartopy. For some reason the x labels are treated differently for basemap but not sure if that is how it should look. It now looks like: 
![Uploading image.png…]()

Reply:
This way we don't have to manually trunk it